csm_web/frontend/src/components/CourseMenu.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Link, Route, Switch } from "react-router-dom";
+import { fetchJSON } from "../utils/api";
+import Course from "./Course";
+import LoadingSpinner from "./LoadingSpinner";
+
+interface CourseType {
+  id: number;
+  name: string;
+  enrollmentOpen: boolean;
+}
+
+interface CourseMenuState {
+  courses: Map<number, CourseType>;
+  loaded: boolean;
+}
+
+interface CourseMenuProps {
+  match: { path: string };
+}
+
+export default class CourseMenu extends React.Component<CourseMenuProps> {
+  state: CourseMenuState = { courses: null, loaded: false };
+
+  componentDidMount() {
+    fetchJSON("/courses").then(courses => {
+      // We use a Map here instead of an object because we want the entries() iterator to reflect the order of insertion,
+      // which in turn reflects the sorting order returned by the backend
+      const coursesById = new Map();
+      for (const course of courses) {
+        coursesById.set(course.id, course);
+      }
+      this.setState({ courses: coursesById, loaded: true });
+    });
+  }
+
+  render() {
+    const { path } = this.props.match;
+    const { loaded, courses } = this.state;
+    const show_enrollment_times = true;
+
+    const enrollment_menu = (
+      <React.Fragment>
+        <h3 className="page-title center-title">Which course would you like to enroll in?</h3>
+        {!loaded ? (
+          <LoadingSpinner id="course-menu-loading-spinner" />
+        ) : (
+          <div id="course-menu">
+            {Array.from(courses.entries()).map(([id, course]) => (
+              <Link
+                className={"csm-btn" + (course.enrollmentOpen ? "" : " disabled")}
+                to={course.enrollmentOpen ? `${path}/${id}` : "#"}
+                key={id}
+              >
+                {course.name}
+              </Link>
+            ))}
+          </div>
+        )}
+      </React.Fragment>
+    );
+
+    const enrollment_times = (
+      <React.Fragment>
+        <h3 className="page-title center-title">Enrollment will open:</h3>
+        <p className="center-title">Wednesday 9/8 at 7 PM for CS61B, CS88</p>
```

Comment:
consider leading with the class name instead, since that's probably how people will instinctively try to look, e.g.
```
**CS88:** Wednesday 9/8 @ 7PM PDT
**CS61A:** Friday 9/10 @ 7PM PDT
...
```
also consider bolding/coloring course names for visual clarity

Reply:
Oops I just saw this; I modified the list to be a grid with a vertical border between course names and times in the latest commit. I can change it over or make any other modifications if needed too.